tests/ProportionalDerivativeControllerTest.cpp

```diff
@@ -107,3 +107,56 @@ TEST_CASE("Proportional Derivative Controller [SE(3)]")
     auto error = state.compose(desiredState.inverse()).log();
     REQUIRE(error.coeffs().norm() < 1e-4);
 }
+
+TEST_CASE("Proportional Derivative Controller [R3]")
+{
+    manif::R3d desiredState, state;
+    desiredState.setRandom();
+    state.setRandom();
+    manif::R3d::Tangent stateDerivative = Eigen::Vector3d::Zero();
+
+    const manif::R3d::Tangent desiredStateDerivative = Eigen::Vector3d::Zero();
+    const auto feedForward = Eigen::Vector3d::Zero();
+
+    // Initialize the controller
+    ProportionalDerivativeControllerR3d controller;
+    constexpr double kp = 10;
+    const double kd = 2 * std::sqrt(kp);
+    controller.setGains({kp, kd});
+    controller.setDesiredState({desiredState, desiredStateDerivative});
+    controller.setFeedForward(feedForward);
+
+    // Test the controller
+    constexpr double dT = 0.01;
+    constexpr std::size_t numberOfIteration = 1e3;
+    for (std::size_t i = 0; i < numberOfIteration; i++)
+    {
+        controller.setState({state, stateDerivative});
+        controller.computeControlLaw();
+        auto controlOutput = controller.getControl();
+
+        // In this specific case the dynamics of the system is a simple double integrator.
+        // and the controller is a simple PD controller in R3.
+        // Let's check the last statement
+
+        Eigen::Vector3d expectedControlOutput = feedForward
+                                                + kd * (desiredStateDerivative.coeffs() - stateDerivative.coeffs())
+                                                + kp * (desiredState.coeffs() - state.coeffs());
+
+        REQUIRE(expectedControlOutput.isApprox(controlOutput.coeffs()));
+
+        // Propagate the system dynamics
+        manif::R3d::Tangent stateDerivativeDT = stateDerivative.coeffs() * dT;
+        state = stateDerivativeDT + state;
```

Comment:
Here, one should not worry about the order or the frame in which state derivative is expressed. Because commutativity applies.

Reply:
yep exactly